groups/bsl/bslstl/bslstl_optional.t.cpp

```diff
@@ -758,120 +980,11218 @@ struct UsesAllocatorArgT<MyClass2a> : bsl::true_type {
+                                       VA2,
+                                       MoveUtil::move(VA3),
+                                       VA4,
+                                       MoveUtil::move(VA5),
+                                       VA6),
+                                      &oa);
+        TEST_MAKE_OPTIONAL_WITH_ALLOC((bsl::in_place,
+                                       VA1,
+                                       MoveUtil::move(VA2),
+                                       VA3,
+                                       MoveUtil::move(VA4),
+                                       VA5,
+                                       MoveUtil::move(VA6)),
+                                      (bsl::allocator_arg,
+                                       &oa,
+                                       VA1,
+                                       MoveUtil::move(VA2),
+                                       VA3,
+                                       MoveUtil::move(VA4),
+                                       VA5,
+                                       MoveUtil::move(VA6)),
+                                      &oa);
+
+        TEST_MAKE_OPTIONAL_WITH_ALLOC((bsl::in_place,
+                                       MoveUtil::move(VA1),
+                                       VA2,
+                                       MoveUtil::move(VA3),
+                                       VA4,
+                                       MoveUtil::move(VA5),
+                                       VA6,
+                                       MoveUtil::move(VA7)),
+                                      (bsl::allocator_arg,
+                                       &oa,
+                                       MoveUtil::move(VA1),
+                                       VA2,
+                                       MoveUtil::move(VA3),
+                                       VA4,
+                                       MoveUtil::move(VA5),
+                                       VA6,
+                                       MoveUtil::move(VA7)),
+                                      &oa);
+        TEST_MAKE_OPTIONAL_WITH_ALLOC((bsl::in_place,
+                                       VA1,
+                                       MoveUtil::move(VA2),
+                                       VA3,
+                                       MoveUtil::move(VA4),
+                                       VA5,
+                                       MoveUtil::move(VA6),
+                                       VA7),
+                                      (bsl::allocator_arg,
+                                       &oa,
+                                       VA1,
+                                       MoveUtil::move(VA2),
+                                       VA3,
+                                       MoveUtil::move(VA4),
+                                       VA5,
+                                       MoveUtil::move(VA6),
+      
```

Comment:
It's odd that we don't use the template argument. I think we should.

Reply:
changed the name of template argument to just TYPE, as discussed in emails. Applied throughout the file.